asciidoctorj-core/src/main/java/org/asciidoctor/cli/AsciidoctorInvoker.java

```diff
@@ -100,13 +103,37 @@ private void setVerboseLevel(AsciidoctorCliOptions asciidoctorCliOptions) {
     }
 
     private Asciidoctor buildAsciidoctorJInstance(AsciidoctorCliOptions asciidoctorCliOptions) {
-        Asciidoctor asciidoctor;
-        if(asciidoctorCliOptions.isLoadPaths()) {
-         asciidoctor = JRubyAsciidoctor.create(asciidoctorCliOptions.getLoadPaths());   
-        } else {
-            asciidoctor = JRubyAsciidoctor.create();
+        ClassLoader oldTccl = Thread.currentThread().getContextClassLoader();
+        try {
+            if (asciidoctorCliOptions.isClassPaths()) {
+                URLClassLoader tccl = createUrlClassLoader(asciidoctorCliOptions.getClassPaths());
+                Thread.currentThread().setContextClassLoader(tccl);
+            }
+            Asciidoctor asciidoctor;
+            if (asciidoctorCliOptions.isLoadPaths()) {
+                asciidoctor = JRubyAsciidoctor.create(asciidoctorCliOptions.getLoadPaths());
+            } else {
+                asciidoctor = JRubyAsciidoctor.create();
+            }
+            return asciidoctor;
+        } finally {
+            Thread.currentThread().setContextClassLoader(oldTccl);
+        }
+    }
+
+    private URLClassLoader createUrlClassLoader(List<String> classPaths) {
```

Comment:
What I am wondering is whether  `--cp /path/to/dirFullOfJars`, which is what I think you implemented, or whether `--cp /path/tp/myconverter.jar` is the better cmdline interface. The latter might be more Java-like, so more understandable for JVM people, whereas the first might be more friednly towards less-technical people.